src/test/basics/ProgressDisplay_test.cpp

```diff
@@ -0,0 +1,186 @@
+            std::cout.rdbuf(consoleBuffer_);
+
+            std::stringstream ss;
+
+            ss << "\r[";
+            for (int j = 0; j <= i; ++j)
+                ss << "#";
+
+            for (int j = i + 1; j < progress.displayWidth_; ++j)
+                ss << " ";
+
+            ss << "]" << std::fixed << std::setprecision(2)
+               << (100 * (i + 1.0)) / expectedCount << "% [" << countedItemName
+               << " " << i + 1 << " of " << expectedCount << "]";
+
+            if (i == expectedCount - 1)
+                ss << std::endl;
+
+            BEAST_EXPECT(ss.str() == text);
+        }
+        std::cout.rdbuf(consoleBuffer_);
+    }
+
+    void
+    testLargeExpectedCount(std::string const& testName)
+    {
+        testcase(testName);
+
+        while (captureBuffer_.str().find(testName) == std::string::npos)
+            std::this_thread::sleep_for(std::chrono::milliseconds(100));
+
+        std::cout.rdbuf(consoleBuffer_);
+        std::cout << captureBuffer_.str();
+
+        reset();
+
+        auto const expectedCount = 80;
+        auto const countedItemName = "Item";
+        auto const initialMessage = "Testing Progress Display";
+
+        ProgressDisplay progress(
+            expectedCount, countedItemName, initialMessage);
+        std::cout.rdbuf(consoleBuffer_);
+
+        {
+            std::stringstream ss;
+
+            ss << std::endl
+               << std::string(std::strlen(initialMessage), '=') << std::endl
+               << initialMessage << std::endl
+               << std::string(std::strlen(initialMessage), '=') << std::endl;
+
+            ss << "\r[";
+            for (int i = 0; i < progress.displayWidth_; ++i)
+                ss << " ";
+
+            ss << "]0.00% [" << countedItemName << " " << 0 << " of "
+               << expectedCount << "]";
+
+            BEAST_EXPECT(ss.str() == captureBuffer_.str());
+        }
+
+        reset();
+        for (int i = 0; i < expectedCount; ++i, reset())
+        {
+            ++progress;
+            std::string text = captureBuffer_.str();
+            std::cout.rdbuf(consoleBuffer_);
+
+            std::stringstream ss;
+
+            ss << "\r[";
+            for (int j = 1; j <= i; j += 2)
+                ss << "#";
+
+            for (int j = (i + 1) / 2; j < progress.displayWidth_; ++j)
+                ss << " ";
+
+            ss << "]" << std::fixed << std::setprecision(2)
+               << (100 * (i + 1.0)) / expectedCount << "% [" << countedItemName
+               << " " << i + 1 << " of " << expectedCount << "]";
+
+            if (i == expectedCount - 1)
+                ss << std::endl;
+
+            BEAST_EXPECT(ss.str() == text);
+        }
+        std::cout.rdbuf(consoleBuffer_);
+    }
+
+    void
+    run() override
+    {
+        reset();
+        testTrivialExpectedCount("Trivial Expected Count");
+
+        reset();
+        testLargeExpectedCount("Large Expected Count");
```

Comment:
Since these functions only have one calling location, don't pass the test name. Use a local variable in the function.

Reply:
Removed